New Folder Modal: hide folder templates for disabled languages

sharon-wang opened the pull request:
### Summary

- addresses https://github.com/posit-dev/positron/issues/7414
- hides the folder template buttons for languages that are disabled via `interpreters.startupBehavior`
- added/updated/reorganized New Folder Flow tests and helpers
    - addresses test comments from https://github.com/posit-dev/positron/pull/7696
- see these docs for how to set per-language startup behaviour:
    - https://positron.posit.co/interpreters.html#configuring-interpreter-startup
    - https://docs.posit.co/ide/server-pro/admin/positron_sessions/interpreter_settings.html#per-language-startup-behavior 

### Release Notes

#### Bug Fixes

- Folder templates for disabled languages no longer appear in New Folder from Template modal (#7414)

### QA Notes

@:new-folder-flow @:web @:win @:r-pkg-development @:interpreter

#### new tests
- added tests for Empty Folder New Folder Flow, which includes the default case as well as a test case for `"interpreters.startupBehavior": "disabled"`

#### reorganization of tests
- moved New Folder Flow helpers to test/e2e/tests/new-folder-flow/helpers/new-folder-flow.ts
- separated tests into
    - Empty: test/e2e/tests/new-folder-flow/new-folder-flow-empty.test.ts
    - R: test/e2e/tests/new-folder-flow/new-folder-flow-r.test.ts
    - Jupyter: test/e2e/tests/new-folder-flow/new-folder-flow-jupyter.test.ts
    - Python: test/e2e/tests/new-folder-flow/new-folder-flow-python.test.ts
    - interpreter startup behaviour tests across folder templates: test/e2e/tests/new-folder-flow/new-folder-flow-folder-templates.test.ts

#### test cases
- if `"interpreters.startupBehavior": "disabled"`, only Empty Project is available
- if `"[r]": { "interpreters.startupBehavior": "disabled" }`, R Project is unavailable but everything is is available
- if `"[python]": { "interpreters.startupBehavior": "disabled" }`, Python and Jupyter projects are unavailable but everything is is available

sharon-wang committed:
only show folder templates for non-disabled language templates

sharon-wang committed:
add new folder flow test for `interpreters.startupBehavior` disabled

sharon-wang committed:
reorganize new folder tests and add default empty folder test

sharon-wang commented on test/e2e/tests/new-folder-flow/new-folder-flow-r-jupyter.test.ts:
separated into `new-folder-flow-jupyter.test.ts` and `new-folder-flow-r.test.ts`

sharon-wang commented on test/e2e/tests/new-folder-flow/new-folder-flow-empty.test.ts:
```diff
@@ -0,0 +1,49 @@
+	});
+
+	test('Only Empty Project template shows when interpreter startup behavior is disabled', async function ({ app, workspaceSettings }) {
+		// Disable startup behavior for all interpreters
+		await app.workbench.settings.removeWorkspaceSettings(['interpreters.startupBehavior']);
+		await workspaceSettings.set([['interpreters.startupBehavior', '"disabled"']]);
```
Should we open an issue for updating the testing infra to include the ability to set language-scoped settings? It would allow us to add tests for R and Python/Jupyter when `interpreters.startupBehavior` is set at the python and r language scopes.

testlabauto replied:
Yes, that would be good.  Right now you can get one more deeply nested setting in using string:JSON where JSON is a string containing whatever JSON structure, but as soon as you try two of them, it would be broken because it wouldn't know they needed to merge and would make a broken JSON document.

sharon-wang commented on test/e2e/pages/newFolderFlow.ts:
```diff
@@ -32,7 +32,10 @@ export class NewFolderFlow {
 		await this.setFolderTemplate(options.folderTemplate);
 		await this.setFolderNameLocation(options);
-		await this.setConfiguration(options);
+
+		if (options.folderTemplate !== FolderTemplate.EMPTY_PROJECT) {
+			await this.setConfiguration(options);
+		}
```
A couple additions in this file to handle the Empty Project flow, which does not have a configuration step, and thus has a <kbd>Create</kbd> button at the folder name/location step instead of a Next button.

testlabauto commented on test/e2e/tests/new-folder-flow/helpers/new-folder-flow.ts:
```diff
@@ -0,0 +1,81 @@
+	await test.step(`Verify folder created`, async () => {
+		await expect(app.code.driver.page.locator('#top-action-bar-current-working-folder')).toHaveText(folderName, { timeout: 60000 }); // this is really slow on windows CI for some reason
+
+		const consoleSymbol = folderTemplate === FolderTemplate.R_PROJECT ? '>' : '>>>';
```
Should this maybe be skipped when folderTemplate is EMPTY_PROJECT?

sharon-wang replied:
yes! This is now fixed

sharon-wang committed:
split console check from folder creation check

don't check for console ready in Empty Folder test, since the user doesn't select an interpreter in the New Folder Flow

sharon-wang committed:
add interpreter behavior disabled tests for all folder templates

sharon-wang committed:
ensure git files show up in explorer

sharon-wang commented on test/e2e/tests/new-folder-flow/helpers/new-folder-flow.ts:
```diff
@@ -0,0 +1,81 @@
+export async function verifyGitFilesArePresent(app: Application) {
+	await test.step('Verify that the .git files are present', async () => {
+		const files = app.code.driver.page.locator('.monaco-list > .monaco-scrollable-element');
+		await expect(files.getByText('.git')).toBeVisible({ timeout: 50000 });
```
I'm not sure how this was working 🤔 if the test/user hasn't removed the default configuration `**/.git` from the `file.excludes` setting, this will always fail. I've updated this helper function to modify the setting temporarily so that `.git` files aren't excluded from the explorer, so that this check can pass.

sharon-wang commented:
Windows e2e failure `Error: EPIPE: broken pipe, write` is a known test pipeline error that is being addressed separately! Will rebase on main once those changes are available.

testlabauto committed:
make waitForTerminalText always use a "canvas approach"

midleman committed:
remove lag in new-folder-flow-templates test

softwarenerd reviewed (dismissed):
I gave everything a test locally and it's all working well for me. Nice job! Thanks for adding the Empty Folder e2e tests.

midleman committed:
consistent naming conventions

midleman committed:
add missing hotkey awaits

midleman committed:
change workspaceSettings fixture scope to test

midleman committed:
make fixture name accurate

sharon-wang committed:
Merge remote-tracking branch 'origin/main' into folder-templates-filter-startup-behavior
